src/main/java/de/tum/in/www1/artemis/repository/ParticipantScoreRepository.java

```diff
@@ -27,7 +28,14 @@
 @Repository
 public interface ParticipantScoreRepository extends JpaRepository<ParticipantScore, Long> {
 
-    void removeAllByExerciseId(Long exerciseId);
+    /**
+     * Delete all participant scores for a given exercise
+     * Note: Only call this method when the exercise is about to be deleted. Otherwise, use {@link #clearAllByResultId(Long)}.
+     * @param exerciseId the exercise id for which to remove all participant scores
+     */
+    @Transactional // required! ok because of delete
```

Comment:
@krusche It seems a normal `Transactional` (without `Propagation.REQUIRES_NEW`) seems to work here as well. At least in my testing.